### DISCUSSION TITLE
Linear movement in Manim

### DISCUSSION THREAD

**Discussion by VladislavKorecky** [ManimCommunity/manim#3275] [Q&A] [answered]
I started learning Manim when I realized that the `move_to()` command doesn't produce linear movement. In the animation I placed below, the circles seem to first accelerate, before deaccelerating again. The red circle was created with `animate.move_to()` and the green circle below with `MoveToTarget`. Both seem to produce the same behavior. I heard that Manim interpolates between the end position and start position to animate, so if I had to guess, Manim uses some sigmoidal interpolation instead of lerp. Is there a way to make the movement linear? Do I have to write my animation or something?

https://github.com/ManimCommunity/manim/assets/55646812/f951dcee-2240-4131-bc1d-091d3640564e

**Comment by uwezi** [comment, marked as answer]
That is the normal behavior in Manim. Animations use a "smooth" timescale which first accelerates and then decelerates by default. If you add the following option to your `self.play()` then a linear timescale will be used: `rate_func=rate_functions.linear`.

More about the different rate functions and their use you will find in the documentation at https://docs.manim.community/en/stable/reference/manim.utils.rate_functions.html?highlight=rate_func

https://github.com/ManimCommunity/manim/assets/8582807/7ae099d7-4168-4df3-9765-8927e124180f

There are plenty of other ways to ensure a pure linear movement or even controlled and accurate accelerated movement, more about which you could learn on the [Discord server](https://docs.manim.community/en/stable/faq/general.html?highlight=discord#where-can-i-find-more-resources-for-learning-manim) 

https://github.com/ManimCommunity/manim/assets/8582807/ec9cc741-8f91-474d-9e01-56e7abe1097e

https://github.com/ManimCommunity/manim/assets/8582807/f1fc6a7d-155d-4984-a011-0c2446885216

**Reply by VladislavKorecky** [reply]
Thanks a lot!
